### PR TITLE
melodic/noetic: add optional move_base interface to base emulator

### PR DESCRIPTION
**Pull request by floweisshardt**

### CODE REVIEW

File: cob_hardware_emulation/scripts/emulation_base.py [hunk 1]
```diff
@@ -158,6 +177,49 @@ def timer_cb(self, event):
...
+    def move_base_simple_callback(self, msg):
+        goal = MoveBaseGoal()
+        goal.target_pose = msg
+        ac_move_base = actionlib.SimpleActionClient(self.move_base_action_name, MoveBaseAction)
+        rospy.loginfo("Waiting for ActionServer: %s", self.move_base_action_name)
+        if not ac_move_base.wait_for_server(rospy.Duration(1)):
+            rospy.logerr("Emulator move_base simple failed because move_base action server is not available")
+            return
+        rospy.loginfo("send goal to %s", self.move_base_action_name)
+        ac_move_base.send_goal(goal)
```
Comment (benmaidel) [inline cob_hardware_emulation/scripts/emulation_base.py:220]: This is calling the own action_server. Can't you just call the internal function `move_base_cb` instead?

**Review by benmaidel — CHANGES_REQUESTED** [review]

File: cob_hardware_emulation/scripts/emulation_base.py (hunk 1 above)
Comment (floweisshardt) [reply cob_hardware_emulation/scripts/emulation_base.py:220]: yes, I did that first. But then I couldn't get the action handle correctly. So the `move_base_cb` complained about setting a result (e.g. succeeded in Line 209) for an invalid action handle.

Comment (benmaidel) [reply cob_hardware_emulation/scripts/emulation_base.py:220]: ah ok. sure, if no action is active the action methods (set_aborted, set_succeeded...) will fail. We could separate the action logic from the application logic into separate methods. But I think that is fine for now...

**Review by benmaidel — APPROVED** [review]

**Comment by floweisshardt** [comment]
seems to work well so far

**Pull request merged**